clang/lib/Sema/SemaType.cpp

```diff
@@ -6070,7 +6070,7 @@ static TypeSourceInfo *GetFullTypeForDeclarator(TypeProcessingState &state,
 ///
 /// The result of this call will never be null, but the associated
 /// type may be a null type if there's an unrecoverable error.
-TypeSourceInfo *Sema::GetTypeForDeclarator(Declarator &D, Scope *S) {
+TypeSourceInfo *Sema::GetTypeForDeclarator(Declarator &D) {
```

Comment:
How does removing this, which appears to not be used, fix the issue?  Or is it only the moved decl search in HandleDeclarator that does that?

Reply:
Removing the unused parameter is more of a "courtesy" to anyone trying to make sense of the changes in the future. This explicitly indicate no dependency on the scope, which I believe to be better than changing the argument value at the call site without further elaboration. 

The commit that fixes the issue does not strictly depend on the commit which removes these unused parameters, but I saw them as being sufficiently related to warrant their inclusion in the same PR. I can move the first commit to it's own PR if you think that's better.

Reply:
that would very much make this easier to review and would be preferential.